### PR TITLE
[DOC] Process doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
RDoc for ::fork.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
My commits say '[DOC]';  why so many tests, I wonder.

**Comment by peterzhu2118** [comment]
> My commits say '[DOC]'; why so many tests, I wonder.

These test are running on the pull request, which uses your pull request title I think. Your PR title doesn't have `[DOC]` in it.

File: process.c [hunk 1]
```diff
@@ -8189,9 +8224,7 @@ rb_clock_gettime(int argc, VALUE *argv, VALUE _)
 
     VALUE unit = (rb_check_arity(argc, 1, 2) == 2) ? argv[1] : Qnil;
     VALUE clk_id = argv[0];
-#ifdef HAVE_CLOCK_GETTIME
     clockid_t c;
-#endif
```
Comment (peterzhu2118) [inline process.c:8194]: Is there a reason why this code was changed?

File: process.c [hunk 2]
```diff
@@ -8419,9 +8452,7 @@ rb_clock_getres(int argc, VALUE *argv, VALUE _)
     timetick_int_t denominators[2];
     int num_numerators = 0;
     int num_denominators = 0;
-#ifdef HAVE_CLOCK_GETRES
     clockid_t c;
-#endif
```
Comment (peterzhu2118) [inline process.c:8424]: Ditto.

File: process.c (hunk 1 above)
Comment (BurdetteLamar) [reply process.c]: Absolutely no idea how this happened (but I should have seen it in the diffs).

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**